controllers/platform/services/services.go

```diff
@@ -198,11 +198,24 @@ func (d DataIndexHandler) MergePodSpec(podSpec corev1.PodSpec) (corev1.PodSpec,
 	return *c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Data Index service specification is not nil
+func (d DataIndexHandler) hasPostgreSQLConfigured() bool {
+	return (d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSQL != nil) ||
+		(d.platform.Spec.Persistence != nil && d.platform.Spec.Persistence.PostgreSQL != nil)
+}
+
 func (d DataIndexHandler) ConfigurePersistence(containerSpec *corev1.Container) *corev1.Container {
-	if d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSql != nil {
+
+	if d.hasPostgreSQLConfigured() {
+		var p *operatorapi.PersistencePostgreSQL
 		c := containerSpec.DeepCopy()
 		c.Image = d.GetServiceImageName(constants.PersistenceTypePostgreSQL)
-		c.Env = append(c.Env, persistence.ConfigurePostgreSqlEnv(d.platform.Spec.Services.DataIndex.Persistence.PostgreSql, d.GetServiceName(), d.platform.Namespace)...)
+		if d.platform.Spec.Services.DataIndex.Persistence != nil && d.platform.Spec.Services.DataIndex.Persistence.PostgreSQL != nil {
```

Comment:
same

Reply:
Not sure because adding it to `hasPostgreSQLConfigured()` already validates it in line 209:
https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/372/files/95b390b7469f45c3cfad661dd5d1f845757fcdc1#diff-9168d999a0cd028523626009a2534c70de62a6e4322770f8bc21d8fc01fc2b83R209